haskell/private/actions/link.bzl

```diff
@@ -108,6 +108,16 @@ def link_binary(
             print("WARNING: dynamic linking and profiling don't mix. Omitting -dynamic.\nSee https://ghc.haskell.org/trac/ghc/ticket/15394")
         else:
             args.add_all(["-pie", "-dynamic"])
+    elif not hs.toolchain.is_darwin and not hs.toolchain.is_windows:
+        # GHC compiles the `main` function during linking with non-position
+        # independent code. On some more recent Linux distributions (e.g.
+        # Ubuntu 18.04) `gcc` defaults to linking with `-pie`, so we need
```

Comment:
Is there an upstream ticket for why GHC is unable to produce PIEs?

Reply:
There's a note in the GHC code base that explains this a bit further: https://gitlab.haskell.org/ghc/ghc/-/blob/d0bab2e3419e49cdbb1201d4650572b57f33420c/compiler/main/DynFlags.hs#L5551-5557.
I've added a pointer to it and related GHC tickets to the note in `haskell/private/actions/link.bzl`.